The tests in xattr.c assume an FS with xattr support

The tests in `tests/preload/xattr/xattr.c` expect `setxattr` to work on paths not mounted under pmemfile.
This caused a problem when I had `TEST_DIR=/dev/shm`:
```
$ cf -R preload_xa
Test project /home/tej/code/pmemfile/build
    Start 68: preload_xattr
1/1 Test #68: preload_xattr ....................***Failed    0.09 sec
preload_xattr: setxattr failed: Operation not supported
CMake Error at /home/tej/code/pmemfile/tests/helpers.cmake:105 (message):
  /home/tej/code/pmemfile/build/tests/preload/preload_xattr /dev/shm//xattr
  failed: 3
Call Stack (most recent call first):
  /home/tej/code/pmemfile/tests/preload/xattr/xattr.cmake:45 (execute)
```
This test failure looks exactly as if libpmemfile would return ENOTSUP, but in this case the kernel returned this error code.

--- a/tests/preload/CMakeLists.txt
+++ b/tests/preload/CMakeLists.txt
@@ -29,22 +29,44 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+include(CheckCSourceRuns)
+
+CHECK_C_SOURCE_RUNS("
+#include <sys/xattr.h>
+#include <fcntl.h>
+#include <err.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+int main() {
+	const char *dir_path = \"${TEST_DIR}\";
+	const char *path = \"${TEST_DIR}/dummy_file\";
+
+	(void) mkdir(dir_path, 0700);
+	if (creat(path, 0700) < 0)
+		err(2, \"creat\");
+	if (setxattr(path, \"user.attr1\", \"value\", 5, XATTR_CREATE) < 0)
+		err(3, \"setxattr\");
+	(void) unlink(path);
+
+	return 0;
+}
+" XATTR_AVAILABLE_IN_TEST_DIR)
+
 add_executable(preload_basic basic/basic.c)
 add_executable(preload_config config/config.c)
 add_executable(preload_pool_locking pool_locking/pool_locking.c)
-add_executable(preload_xattr xattr/xattr.c)
 add_executable(preload_unix unix/unix.c)
 
 add_cstyle(tests-preload-basic ${CMAKE_CURRENT_SOURCE_DIR}/basic/basic.c)
 add_cstyle(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 add_cstyle(tests-preload-pool-locking ${CMAKE_CURRENT_SOURCE_DIR}/pool_locking/pool_locking.c)
-add_cstyle(tests-preload-xattr ${CMAKE_CURRENT_SOURCE_DIR}/xattr/xattr.c)
 add_cstyle(tests-preload-unix ${CMAKE_CURRENT_SOURCE_DIR}/unix/unix.c)
 
 add_check_whitespace(tests-preload-basic ${CMAKE_CURRENT_SOURCE_DIR}/basic/basic.c)
 add_check_whitespace(tests-preload-config ${CMAKE_CURRENT_SOURCE_DIR}/config/config.c)
 add_check_whitespace(tests-preload-pool-locking ${CMAKE_CURRENT_SOURCE_DIR}/pool_locking/pool_locking.c)
-add_check_whitespace(tests-preload-xattr ${CMAKE_CURRENT_SOURCE_DIR}/xattr/xattr.c)
 add_check_whitespace(tests-preload-unix ${CMAKE_CURRENT_SOURCE_DIR}/unix/unix.c)
 
 add_library(setumask SHARED setumask.c)
@@ -123,5 +145,11 @@ add_test_generic(config "_outer_via_invalid" $<TARGET_FILE:preload_config> -DTES
 set_tests_properties("preload_config_outer_via_invalid"
 	PROPERTIES PASS_REGULAR_EXPRESSION "EIO")
 
-add_test_generic(xattr "" $<TARGET_FILE:preload_xattr>)
 add_test_generic(unix "" $<TARGET_FILE:preload_unix>)
+
+if(XATTR_AVAILABLE_IN_TEST_DIR)
+	add_executable(preload_xattr xattr/xattr.c)
+	add_cstyle(tests-preload-xattr ${CMAKE_CURRENT_SOURCE_DIR}/xattr/xattr.c)
+	add_check_whitespace(tests-preload-xattr ${CMAKE_CURRENT_SOURCE_DIR}/xattr/xattr.c)
+	add_test_generic(xattr "" $<TARGET_FILE:preload_xattr>)
+endif()

--- a/tests/preload/xattr/xattr.c
+++ b/tests/preload/xattr/xattr.c
@@ -64,20 +64,20 @@ main(int argc, char *argv[])
 	if (size != -1)
 		err(2, "attr1 already exists");
 
-	const char *lorem =
+	static const char lorem[] =
 		"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.";
-	size_t lorem_len = strlen(lorem);
 
-	if (setxattr(path, "user.attr1", lorem, lorem_len, XATTR_CREATE) != 0)
+	if (setxattr(path, "user.attr1", lorem, sizeof(lorem),
+			XATTR_CREATE) != 0)
 		err(3, "setxattr failed");
 
 	size = getxattr(path, "user.attr1", value, sizeof(value));
 	if (size < 0)
 		err(4, "attr1 is empty");
-	if ((size_t)size != lorem_len)
+	if ((size_t)size != sizeof(lorem))
 		errx(5, "attr1 has unexpected value %ld", size);
 
-	if (strcmp(lorem, value) != 0)
+	if (memcmp(lorem, value, sizeof(lorem)) != 0)
 		err(6, "unexpected attr1 value: %s", value);
 
 	sprintf(path, "%s/mount_point/../file", argv[1]);
@@ -86,23 +86,23 @@ main(int argc, char *argv[])
 	size = getxattr(path, "user.attr1", value, sizeof(value));
 	if (size < 0)
 		err(7, "attr1 is empty (2)");
-	if ((size_t)size != lorem_len)
+	if ((size_t)size != sizeof(lorem))
 		errx(8, "attr1 has unexpected value %ld (2)", size);
 
-	if (strcmp(lorem, value) != 0)
+	if (memcmp(lorem, value, sizeof(lorem)) != 0)
 		err(9, "unexpected attr1 value: %s", value);
 
-	if (setxattr(path, "user.attr1", "meh", 3, XATTR_REPLACE) != 0)
+	if (setxattr(path, "user.attr1", "meh", 4, XATTR_REPLACE) != 0)
 		err(10, "setxattr failed (2)");
 
 	memset(value, 0, sizeof(value));
 	size = getxattr(path, "user.attr1", value, sizeof(value));
 	if (size < 0)
 		err(11, "attr1 is empty (3)");
-	if ((size_t)size != 3)
+	if ((size_t)size != 4)
 		errx(12, "attr1 has unexpected value %ld (3)", size);
 
-	if (strcmp("meh", value) != 0)
+	if (memcmp("meh", value, 4) != 0)
 		err(13, "unexpected attr1 value: %s (2)", value);
 
 	memset(value, 0, sizeof(value));

